### PR TITLE
Test the nixos module

### LINKED ISSUE #114: Automatic tests for the nixos module
We currently support an increasing number of ways to use the nixos module, and it's getting a bit unwieldy to test the different paths. Having automatic tests for the various variants would be great.

### PR DESCRIPTION
**Pull request by antifuchs**
This PR fixes #114.

Here, we add two VM-based nixos module tests, one for a basic listening configuration and one for a custom certificate config.

This required stubbing out the actual functionality from tsnsrv - to that end, we introduce a second package `tsnsrvCmdLineValidator`, which validates commandline arguments and if they check out, waits forever (like a real tsnsrv would); that makes it compatible for usage in the systemd unit, which we use in tests.

I would love to test the actual functionality against a really-running headscale, but for now, this will do and catch things like #112.

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -122,6 +124,61 @@
         };
         formatter = pkgs.alejandra;
 
+        checks = let
+          nixos-lib = import "${nixpkgs}/nixos/lib" {};
```
Comment (SuperSandro2000) [inline flake.nix:128]:
```suggestion
          nixpkgs-lib = nixpkgs.lib;
```

lib doesn't require an import and nixos-lib is typical something different https://github.com/NixOS/nixpkgs/tree/master/nixos/lib

File: flake.nix [hunk 2]
```diff
@@ -122,6 +124,61 @@
...
+              testConfig,
+              testScript,
+            }:
+              nixos-lib.runTest {
+                name = "tsnsrv-nixos";
+                hostPkgs = pkgs;
+
+                defaults.services.tsnsrv.enable = true;
+                defaults.services.tsnsrv.defaults.package = config.packages.tsnsrvCmdLineValidator;
+                defaults.services.tsnsrv.defaults.authKeyPath = "/dev/null";
```
Comment (SuperSandro2000) [inline flake.nix:143]: nixos has snakeoil certs available for testing https://github.com/NixOS/nixpkgs/tree/master/nixos/tests/common/acme/server

Comment (antifuchs) [reply flake.nix:143]: Nice! I think I want to replace these tests with something real that uses headscale anyway - the stub commandline validator doesn't feel great, but it does the trick for now (and it doesn't really care about the existence of files). I'll keep the certs in mind when I redo the tests!

File: flake.nix (hunk 1 above)
Comment (antifuchs) [reply flake.nix:128]: This doesn't have the `runTest` attribute; I think this really might need the nixos/lib import?

Comment (SuperSandro2000) [reply flake.nix:128]: Derp, I've overread this 😂 You're right

**Pull request merged**